processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -35,6 +37,8 @@ public TopNMapFn(
   {
     this.query = query;
     this.topNAlgorithm = topNAlgorithm;
+    this.dimension = query.getDimensionSpec().getDimension();
+    this.extractionFn = TopNQueryEngine.canApplyExtractionInPost(query) ? null : query.getDimensionSpec().getExtractionFn();
```

Comment:
The rewrite only happens if results are merged. If the segment servicing node is queried directly, it may not go through the merge process (at least with the pre/post merge rewriting)